### DISCUSSION TITLE
'Please be sure to set the "credential_secret" option' error when starting in HA OS

### DISCUSSION THREAD

**Discussion by cadamwil** [hassio-addons/addon-node-red#1460] [Q&A] [answered]
I get the following in the log when I attempt to start the plugin.

`s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/00-banner.sh
-----------------------------------------------------------
 Add-on: Node-RED
 Flow-based programming for the Internet of Things
-----------------------------------------------------------
 Add-on version: 13.4.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 9.0  (amd64 / qemux86-64)
 Home Assistant Core: 2022.9.7
 Home Assistant Supervisor: 2022.09.1
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
cont-init: info: /etc/cont-init.d/00-banner.sh exited 0
cont-init: info: running /etc/cont-init.d/01-log-level.sh
cont-init: info: /etc/cont-init.d/01-log-level.sh exited 0
cont-init: info: running /etc/cont-init.d/customizations.sh
cont-init: info: /etc/cont-init.d/customizations.sh exited 0
cont-init: info: running /etc/cont-init.d/nginx.sh
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
cont-init: info: running /etc/cont-init.d/node-red.sh
[12:08:55] FATAL: 
[12:08:55] FATAL: Configuration of this add-on is incomplete.
[12:08:55] FATAL: 
[12:08:55] FATAL: Please be sure to set the "credential_secret" option.
[12:08:55] FATAL: 
[12:08:55] FATAL: The credential secret is an encryption token, much like
[12:08:55] FATAL: a password, that is used by Node-RED for encrypting
[12:08:55] FATAL: credentials you put into Node-RED.
[12:08:55] FATAL: 
[12:08:55] FATAL: Just like a password, a credential secret can be
[12:08:55] FATAL: anything you like. Just be sure to store it somewhere
[12:08:55] FATAL: safe for later, e.g., in case of a recovery.
[12:08:55] FATAL: 
cont-init: info: /etc/cont-init.d/node-red.sh exited 1
cont-init: warning: some scripts exited nonzero
s6-rc: warning: unable to start service legacy-cont-init: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
prog: fatal: stopping the container.
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped`

Any help would be greatly appriciated.

**Comment by frenck** [comment, marked as answer]
Please... read the documentation and follow the installation steps.

You missed step 3.

**Reply by Richardrsb** [reply]
Nice answer. Step 3 says "start node-red" It won't start. The log mentions "Configuration of this add-on is incomplete."
If i start the configuration i get the message that there is no configuration.
Realize there are users new to this. The log tells you nothing.

**Reply by frenck** [reply]
@Richardrsb Seems like you have even a harder time reading than other people, trying to correct me there and give out some kind of punch?

Let me paste a screenshot:

![CleanShot 2023-10-22 at 22 23 50@2x](https://github.com/hassio-addons/addon-node-red/assets/195327/0079d112-f94e-4e4e-a8e1-1d020c918b57)

> Step 3 says "start node-red"

As per screenshot. It **does not** say that.

> Realize there are users new to this

Sorry, I cannot help people who are new to reading. I'm not an elementary school.

../Frenck

**Reply by nexnox** [reply]
![image](https://github.com/hassio-addons/addon-node-red/assets/20271911/2e6b099d-3bdb-463b-8e2a-c8b6428db95b)
the documentation page on Home Assistant doesn't mention it and has the other Step 3 mentioned.

**Comment by Richardrsb** [comment]
I don't try to give you a punch. The problem is , that for some users this is not so obvious because for some reason step 3 is not so obvious as you might think.  My guess is you assume too much considering other installations.

In my case:

The installation runs fine, after that: nothing. Where does one set a credential secret? (Your Step 3)
How does one configure node-red, if the configuration button tells you there is nothing to configure?
Starting Node Red result in the log above with all the fatal messages, and it won't start.

Thats it in a nutshell. What is obvious for you, *CAN* be very confusing for others if an installation not behaves
as one might expect.

Going through the github issues i learned that i am not alone.

Cheers.

Richard

Oh, and as for reading: this screenshot clearly states step 3: Start the Node-Red addon.
I hope this proves i don't need to learn to read.

![Screenshot 2023-10-23 at 09 10 09](https://github.com/hassio-addons/addon-node-red/assets/125134836/d02b862d-7978-4f9c-a217-21ae80848b77)

**Reply by frenck** [reply]
> The problem is , that for some users this is not so obvious because for some reason step 3 is not so obvious as you might think.

Then this add-on is not for them. It is that simple.

> My guess is you assume too much considering other installations.

Which other installations? This is only aimed at a single one.

But honestly, if these very simple instructions are problematic for you, please uninstall this add-on. The rest of the journey will be a horror story in that case.

../Frenck

**Reply by Richardrsb** [reply]
Frenck,

Anyone can follow the simple instructions in both scenarios. But if the result is different than what you expect, you are clearly unable to imagine that for that user it gets confusing and gets stuck by a simple instruction as "Set a credential_secret" if there is no way to set that aforementioned credential_secret, and the configuration window tells the user there is no configuration.

Maybe you are right: this is not for everyone.

You are clearly a brilliant coder, but still unable to even imagine that some users have trouble if something is not going as -simple- advertised.

And as for reading and understanding: with "other installations" i refer to all users who install nod-red. Do i really need to explain that?

I am not trying to point fingers nor to punch you. I really appreciate your (and others) work in the community. I am not a coder, but i am also not stupid.

If i want a cappuccino from my newly installed coffeemaker, and the simple instructions for cappuccino tells me to press the cappuccino button, and there is no cappuccino button to find on the entire coffeemaker then i am confused as a new user.

Ok, leave it as is. Just get the black coffee, and add milk later? Or seek help in the community. Then, the manufacturer tells me to press the freaking cappuccino button as per instructions. While there is none to press on my particular coffeemaker.

Just my humble opinion. 

Regards,

Richard

**Reply by frenck** [reply]
Thanks.

**Reply by Richardrsb** [reply]
Frenck, 

Just a follow up:
After 3 reinstalls, Node-Red showed up. I don't know why it did not in the earlier attempts. At a certain point i was presented the configuration page, and was able to set the aforementioned  required credentials in the configuration window.

After some study (I am new to Node-Red) i was able to insert a modbus server (Slave) which gracefully serves the required data to my eGauge energy meter.  Works perfect.

Thanks for your efforts in HA and Node-Red. 

Regards, Richard

![Screenshot 2023-11-02 at 09 17 43](https://github.com/hassio-addons/addon-node-red/assets/125134836/be32a9c2-acdf-4a3b-978b-972968821b57)

**Comment by wrmacj** [comment]
Can you help with the install instructions re the message in the logs, copied below.  I cannot find details on how to find the 'credentialSecret'? 
According to the logs the settings file is [info] Settings file  : **/etc/node-red/config.js**

ssh, into my installation and **/etc/node-red/config.js** does not exist anywhere in the file system.

The options under configuration also do not have a reference to 'credentialSecret'

theme: default
http_node:
  username: ""
  password: ""
http_static:
  username: ""
  password: ""
ssl: true
certfile: fullchain.pem
keyfile: privkey.pem
system_packages: []
npm_packages: []
init_commands: []

➜  /etc **find / -name config.js**
find: /proc/776/task/776/fdinfo: Permission denied
find: /proc/776/fdinfo: Permission denied
➜  /etc **find / -name node-red**
find: /proc/776/task/776/fdinfo: Permission denied
find: /proc/776/fdinfo: Permission denied
➜  /etc

---------------------------------------------------------------------
Your flow credentials file is encrypted using a system-generated key.

If the system-generated key is lost for any reason, your credentials
file will not be recoverable, you will have to delete it and re-enter
your credentials.

You should set your own key using the 'credentialSecret' option in
your settings file. Node-RED will then re-encrypt your credentials
file using your chosen key the next time you deploy a change.
---------------------------------------------------------------------

**Reply by Richardrsb** [reply]
What helped for me was to uninstall Node-Red, and re-install.  Perhaps booting HA in between. I am not sure how and why, but after the third attempt I got the configuration screen presented. Read above. Hope this helps.

**Comment by wrmacj** [comment]
Hmmmmm Asked and answered....all that was required was to switch the config to yaml and add the following to the bottom of the file.
credentialSecret: YoursEcret

What is frustrating is that you are recommended to do it if you dig into the log file, you are reminded to do it in the debug log but it is not suggested or given decent instructions on how to accomplish the task.

Could we get the option to add the credentialSecret option added to the config page.  After I added the credentialSecret entry the option to added it suddenly appeared on the Config page as it always should have been?

After entering the credentialSecret the warning disappeared from the log file.  Definitely an area of opportunity....

![image](https://github.com/hassio-addons/addon-node-red/assets/109797056/c44bcc45-1cc6-4b7d-b990-c9843370549e)

**Reply by frenck** [reply]
If you have suggestions to improve that UI, please feel free to raise a feature request/suggestion for the Home Assistant project. This add-on doesn't provide the UI (only the options).

../Frenck

**Reply by chessmaster82** [reply]
I've got the same issue, "credential secret" no longer shows on the settings page after a fresh install of hassio 11.2
